CMakeLists.txt

```diff
@@ -1,17 +1,19 @@
+set(CMAKE_CXX_STANDARD 11)
```

Comment:
Put this after `cmake_minimum_required`. And also consider using `target_compile_features(pynvjitlink PRIVATE cxx_std_11)` instead, if my understanding of this conversation is correct: https://github.com/rapidsai/cudf/pull/13795#discussion_r1393154975